main/main.go

```diff
@@ -356,6 +359,32 @@ func initSeedGen(ctx context.Context, c client.Client, log *logr.Logger) error {
 		return err
 	}
 
+	// Restore original pull-secret after seed creation.
+	// During seed creation, the pull-secret was removed; it needs to be restored back
+	// to allow the cluster operators to fully recover in the seed cluster.
+	log.Info("Restore original pull-secret after seed creation")
+	dockerConfigJSON, err := os.ReadFile(filepath.Join(common.Host, common.ImageRegistryAuthFile))
+	if err != nil {
+		return fmt.Errorf("failed to read .dockerconfigjson file: %w", err)
+	}
+
+	pullSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      common.PullSecretName,
+			Namespace: "openshift-config",
+		},
+		Data: map[string][]byte{
+			".dockerconfigjson": []byte(base64.StdEncoding.EncodeToString(dockerConfigJSON)),
```

Comment:
I actually sure you don't need to encode it, it works without it too, secret is automatically encoded